torch/signal/windows/windows.py

```diff
@@ -0,0 +1,302 @@
+import torch
+import numpy as np
+
+from torch import Tensor
+from torch._prims_common import is_float_dtype, is_complex_dtype
+from torch._torch_docs import factory_common_args
+from torch.types import _dtype, _device, _layout
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+
+def _add_docstr(*args):
+    r"""Adds docstrings to a given decorated function.
+
+    Specially useful when then docstrings needs string interpolation, e.g., with
+    str.format().
+    REMARK: Do not use this function if the docstring doesn't need string
+    interpolation, just write a conventional docstring.
+
+    Args:
+        args (str):
+    """
+    def decorator(o):
+        o.__doc__ = ""
+        for arg in args:
+            o.__doc__ += arg
+        return o
+    return decorator
+
+
+def _window_function_checks(function_name: str, window_length: int, dtype: _dtype, layout: _layout) -> None:
```

Comment:
Should we rename `window_length -> M`?

Reply:
Yeah, let's just follow SciPy notation everywhere.